### PR TITLE
feat: add proto and domain for search filter

### PR DESCRIPTION
**Pull request by kakcy**
This pull request introduces a new `SearchFilter` feature across multiple files in the project. The changes include the implementation of the `SearchFilter` struct and its associated methods, updates to the protobuf definitions, and the addition of corresponding tests.

### Implementation of `SearchFilter`:

* [`pkg/account/domain/search_filter.go`](diffhunk://#diff-60e96e5c34f4068f4401587fe8461d65f7e07f9318a4ca93de9fac395f97a2a7R1-R59): Added the `SearchFilter` struct and implemented methods `NewSearchFilter`, `UpdateSearchFilter`, and `SetDefaultFilter`.

### Protobuf Definitions:

* [`proto/account/search_filter.proto`](diffhunk://#diff-42b5010b9db6b043de8c3ec5266b279b2ff8151390eb2a75c15e1726368eb49fR1-R33): Defined the `SearchFilter` message and `FilterTargetType` enum.

### Testing:

* [`pkg/account/domain/search_filter_test.go`](diffhunk://#diff-c7cc3826a0418def2ae1dd2774e0d88a6d9fbe735c5082ae3bcaef96578f6483R1-R52): Added tests for `NewSearchFilter`, `SetDefaultFilter`, and `UpdateSearchFilter` methods.

### CODE REVIEW

File: pkg/account/domain/account.go [hunk 1]
```diff
@@ -106,3 +113,61 @@ func (a *AccountV2) Disable() error {
...
+			a.AccountV2.SearchFilters = nil
+			a.UpdatedAt = time.Now().Unix()
+			return nil
+		} else {
+			return errSearchFilterNotFound
+		}
+	}
+	a.AccountV2.SearchFilters = searchFilters
+	a.UpdatedAt = time.Now().Unix()
+	return nil
```
Comment (cre8ivejp) [inline pkg/account/domain/account.go:161]:
```suggestion
	if !isFound {
	    return errSearchFilterNotFound
	}
	a.AccountV2.SearchFilters = searchFilters
	a.UpdatedAt = time.Now().Unix()
	return nil
```

Comment (kakcy) [reply pkg/account/domain/account.go]: I Changed to delete the filter and exit the function when the ID is found without using `isFound`.
https://github.com/bucketeer-io/bucketeer/blob/feature-search-filter-proto-domain/pkg/account/domain/account.go#L140

**Comment by kakcy** [comment]
I have fixed the points raised. Please check when you are free.
https://github.com/bucketeer-io/bucketeer/pull/1197/commits/d85d3de0dcb64b1a6a6cff8ce0ec43053a8cb636

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**